src/lib.rs

```diff
@@ -63,11 +109,30 @@ struct TurtleFormatter<'a, W: Write> {
 impl<'a, W: Write> TurtleFormatter<'a, W> {
     fn fmt_doc(&mut self, node: Node<'_>) -> Result<()> {
         debug_assert_eq!(node.kind(), "turtle_doc");
-
         let mut context = RootContext::Start;
         let mut row = node.start_position().row;
         let mut prefix_buffer: Vec<(Node<'_>, Vec<Node<'_>>)> = Vec::new();
-        for child in Self::iter_children(node)? {
+        let children = if self.options.sort_subjects {
+            let mut sorted = vec![];
+            let mut to_be_sorted = vec![];
+            for child in Self::iter_children(node)? {
+                match child.kind() {
+                    "triples" => {
+                        to_be_sorted.push(child);
+                    }
+                    _ => sorted.push(child),
```

Comment:
We need to take care of multiple things here

1. comments

2. prefixes and base overriding others:
For example
```turtle
@prefix : <http://example1/> .
:s :p :o .
@prefix : <http://example2/> .
:s :p :o .
```

is not the same as

```turtle
@prefix : <http://example1/> .
@prefix : <http://example2/> .
:s :p :o .
:s :p :o .
```

Reply:
Thank you! I think it's fine to fail if there is a prefix or base override, it's definitely not a good practice.

Reply:
If we can handle it correctly, then we might just not say anything. turtlefmt is a formatter, not a linter. WDYT?

Reply:
true!

Reply:
I implemented the `@prefix` and `@base` part discussed here, and went the warning/error way for the comments for now, partly cause it is much easier, and partly because I feel that in some cases, gluing or not of a comment to a part of the code (a term) is impossible to do right/correctly (without  analyzing the content of the comment plus using complex heuristics).
For today, I will leave it at this.